src/graph/backend/elyzor/elyzor_partition_impl.cpp

```diff
@@ -139,11 +139,28 @@ graph::status_t elyzor_partition_impl_t::compile(
         COMPILE_ASSERT(aengine->kind() == graph::engine_kind_t::dnnl_cpu,
                 "Graph compiler backend only supports cpu engine");
 
-        // get executor somehow
+        // TODO: maybe cache ctx and gc in a static variable
+        const dnnl_graph_compiler_context ctx {
+                /*num_threads=*/std::thread::hardware_concurrency(),
+                /*allocator=*/nullptr, /*deallocator=*/nullptr};
+        const dnnl_graph_compiler *gc;
+        WRAP_GC_CALL(dnnl_graph_compiler_create(&ctx, &gc),
+                "Failed to create graph compiler object");
+
+        std::stringstream json_stream;
+        graph::graph_t(copied_ops_).serialize(json_stream);
+
+        std::string json = json_stream.str();
+
+        const dnnl_graph_compiler_executable *exe;
+        WRAP_GC_CALL(dnnl_graph_compiler_compile(gc, json.data(), &exe),
```

Comment:
In case of exception here, gc is never destroyed.

Reply:
rewritten to use shared ptrs for `exe` and `gc` pointers